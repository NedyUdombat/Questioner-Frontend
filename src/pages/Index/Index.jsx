import React from 'react';
import { Link } from 'react-router-dom';

// components
import Navbar from '../../components/presentational/Navbar/Navbar';
import Button from '../../components/presentational/Button/Button';
import UpcomingMeetups from '../../components/container/UpcomingMeetups/UpcomingMeetups';

// styles
import './Index.scss';

const Index = () => {
  return (
    <div>
      <div className="jumbotron">
        <Navbar />
        <div className="hero-image">
          <div className="hero-text">
            <h1>Questioner</h1>
            <p>An application to crowd-source questions for a meetup.</p>
            <Link to="/register">
              <Button
                type="button"
                name="button"
                className="btn btn-dark-alt"
                value="REGISTER"
              />
            </Link>
          </div>
        </div>
      </div>
      <UpcomingMeetups />
      <footer id="non-fixed-footer">
        <p>Made by Edidiong Udombat @ Andela </p>
      </footer>
    </div>
  );
};

export default Index;
