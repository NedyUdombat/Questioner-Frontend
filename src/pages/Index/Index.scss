@import '../../assets/sass/index';

body {
  background-color: $secondary-white;
}

.jumbotron {
  background-image: linear-gradient(rgba(0, 0, 0, .8), rgba(0, 0, 0, .5)), url('https://res.cloudinary.com/nedy123/image/upload/v1558178618/emily-morter-188019-unsplash_gfjg2f.jpg');
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  min-height: 100vh;
  padding: 2% 6%;
};

.hero-image {
  align-items: center;
  display: flex;
  height: 75vh;
  justify-content: center;
  text-align: center;
}

.hero-text {
  color: #fff;
  text-align: center;
}

.hero-text h1 {
  font-family: $bitter;
  font-size: 8rem;
  font-weight: lighter;
}

.hero-text p {
  font-size: 1.125rem;
  margin-top: 1.25rem;
}

.hero-text button {
  margin: auto;
  width: 50%;
}

footer {
  align-items: center;
  background: #000203;
  bottom: 0;
  color: #f2f2f2;
  display: flex;
  height: 5vh;
  justify-content: center;
  padding: 0 20px;
  position: fixed;
  text-align: center;
  width: 100%;
}

#non-fixed-footer {
  position: relative;
}

