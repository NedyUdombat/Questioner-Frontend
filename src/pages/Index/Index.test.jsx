import React from 'react';
import { shallow } from 'enzyme/build';

import Index from './Index';

describe('Index', () => {
  it('should render correctly', () => {
    const component = shallow(<Index />);
    expect(component).toMatchSnapshot();
  });
});
