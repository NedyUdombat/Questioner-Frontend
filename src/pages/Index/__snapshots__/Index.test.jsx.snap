// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should render correctly 1`] = `
<div>
  <div
    className="jumbotron"
  >
    <ConnectFunction />
    <div
      className="hero-image"
    >
      <div
        className="hero-text"
      >
        <h1>
          Questioner
        </h1>
        <p>
          An application to crowd-source questions for a meetup.
        </p>
        <Link
          to="/register"
        >
          <Button
            className="btn btn-dark-alt"
            name="button"
            type="button"
            value="REGISTER"
          />
        </Link>
      </div>
    </div>
  </div>
  <ConnectFunction />
  <footer
    id="non-fixed-footer"
  >
    <p>
      Made by Edidiong Udombat @ Andela 
    </p>
  </footer>
</div>
`;
